Tests/RegexTests/ParseTests.swift

```diff
@@ -111,6 +117,23 @@ extension RegexTests {
       .leftSquareBracket, .rightSquareBracket,
       .setOperator(.doubleAmpersand), .rightSquareBracket,
       .rightSquareBracket, "&", "&")
+
+    lexTest("$\\A\\B[\\A\\B$]", .anchor(.lineEnd), .anchor(.stringStart),
+            .anchor(.nonWordBoundary), .leftSquareBracket, esc("A"), esc("B"),
+            "$", .rightSquareBracket)
+
+    let specialChars = [.tab, .carriageReturn, .formFeed, .bell, .escape]
+      .map(Token.specialCharEscape)
+
+    lexTest("\\t\\r\\f\\a\\e[\\t\\r\\f\\a\\e]",
+            specialChars + [.leftSquareBracket] + specialChars +
+            [.rightSquareBracket])
+
+    // \b is a word boundary outside of a character class, otherwise it's
+    // backspace.
+    lexTest("[\\b]\\b", .leftSquareBracket, .specialCharEscape(.backspace),
+            .rightSquareBracket, .anchor(.wordBoundary))
+    lexTest("[\\b", .leftSquareBracket, .specialCharEscape(.backspace))
```

Comment:
Is there a test of the 3 meanings of `^`?

Reply:
Not in the same test case, but there are tests that exercise the different meanings